test/functional/JIT_Test/src/jit/test/jitt/array/multianewarray.java

```diff
@@ -0,0 +1,102 @@
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package jit.test.jitt.array;
+
+import org.testng.annotations.Test;
+import org.testng.Assert;	
+import org.testng.AssertJUnit;
+
+@Test(groups = { "level.sanity","component.jit" }, invocationCount=2)
+public class multianewarray {
+	class Vector {
+		long x;
+		long y;
+	}
+
+	private Integer[][] createMultiArray(int firstDimLength, int secondDimLength) {
+		Integer[][] multiarray = new Integer[firstDimLength][secondDimLength];
+		for (int i = 0; i < firstDimLength; i++) {
+			for (int j = 0; j < secondDimLength; j++) {
+				multiarray[i][j] = i * j;
+			}
+		}
+		return multiarray;
+	}
+
+	private Vector[][] createMultiVector(int firstDimLength, int secondDimLength) {
+		Vector[][] multiVector = new Vector[firstDimLength][secondDimLength];
+		for (int i = 0; i < firstDimLength; i++) {
+			for (int j = 0; j < secondDimLength; j++) {
+				multiVector[i][j] = new Vector();
+				multiVector[i][j].x = i * j;
+				multiVector[i][j].y = i * i + j * j;
+			}
+		}
+		return multiVector;
+	}
+
+	private void testMultiVector(Vector[][] inputVector, int dim1Length, int dim2Length) {
+		AssertJUnit.assertEquals(dim1Length, inputVector.length);
+		if (inputVector.length > 0)
+			AssertJUnit.assertEquals(dim2Length, inputVector[0].length);
+
+		for (int i = 0; i < inputVector.length; i++) {
+			for (int j = 0; j < inputVector[0].length; j++) {
+				AssertJUnit.assertEquals(i*j, (inputVector[i][j]).x);
+				AssertJUnit.assertEquals(i*i + j*j, (inputVector[i][j]).y);
+			}
+		}
+	}
+	private void testMultiArray(Integer[][] inputArray, int dim1Length, int dim2Length) {
+		AssertJUnit.assertEquals(dim1Length, inputArray.length);
+		if (inputArray.length > 0)
+			AssertJUnit.assertEquals(dim2Length, inputArray[0].length);
+		for (int i = 0; i < inputArray.length; i++) {
+			for (int j = 0; j < inputArray[0].length; j++) {
+				AssertJUnit.assertEquals(i*j, (inputArray[i][j]).intValue());
+			}
+		}
+	}
+	@Test
+	public void testmultianewarray() {
+		Integer[][] inputArray;
+		Vector[][] inputVector;
+		for (int i = 0; i < 10000; i++) {
+			inputVector = createMultiVector(0, 0);
+			testMultiVector(inputVector, 0, 0);
+			inputArray = createMultiArray(0, 0);
+			testMultiArray(inputArray, 0, 0);
+		}
+		for (int i = 0; i < 10000; i++) {
+			inputVector = createMultiVector(i, 0);
+			testMultiVector(inputVector, i, 0);
+			inputArray = createMultiArray(i, 0);
+			testMultiArray(inputArray, i, 0);
+		}
+		for (int i = 0; i < 10000; i++) {
```

Comment:
I think the 10000 here may be causing timeout issues on Power. For example see here:
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/95/console

```
16:28:11  variation: -Xrs -Xjit:noJitUntilMain,count=0,assumeStrictFP,optlevel=warm,gcOnResolve,rtResolve -verbose:stackwalk=0 -Xdump
16:28:11  JVM_OPTIONS:  -Xrs -Xjit:noJitUntilMain,count=0,assumeStrictFP,optlevel=warm,gcOnResolve,rtResolve -verbose:stackwalk=0 -Xdump 
16:28:11  { itercnt=1; \
16:28:11  mkdir -p "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/../TKG/test_output_16067716828123/jit_jitt_1"; \
16:28:11  cd "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/../TKG/test_output_16067716828123/jit_jitt_1"; \
16:28:11  "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdkbinary/j2sdk-image/bin/java"  -Xrs -Xjit:noJitUntilMain,count=0,assumeStrictFP,optlevel=warm,gcOnResolve,rtResolve -verbose:stackwalk=0 -Xdump  \
16:28:11  -cp "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/../../jvmtest/TestConfig/resources:/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/../TKG/lib/testng.jar:/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/../TKG/lib/jcommander.jar:/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/../../jvmtest/functional/JIT_Test/jitt.jar" \
16:28:11  org.testng.TestNG -d "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/../TKG/test_output_16067716828123/jit_jitt_1" "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/../../jvmtest/functional/JIT_Test/testng.xml" \
16:28:11  -testnames \
16:28:11  arrayTest,\
16:28:11  assemblerTest,\
16:28:11  castingTest,\
16:28:11  cfgTest,\
16:28:11  crashesTest,\
16:28:11  ctresolvesTest,\
16:28:11  exceptionsTest,\
16:28:11  fieldsTest,\
16:28:11  floatsTest,\
16:28:11  gcTest,\
16:28:11  geCastingTest,\
16:28:11  immedOpTest,\
16:28:11  invokeTest,\
16:28:11  mathTest,\
16:28:11  math2Test,\
16:28:11  miscTest,\
16:28:11  os390LinkageTest,\
16:28:11  promotionTest,\
16:28:11  resolvesTest,\
16:28:11  sieveTest,\
16:28:11  transitionsTest \
16:28:11  -groups level.sanity \
16:28:11  -excludegroups d.*.aix_ppc-64_cmprssptrs,d.*.arch.ppc,d.*.os.aix,d.*.bits.64,d.*.generic-all; \
16:28:11  if [ $? -eq 0 ] ; then echo ""; echo "jit_jitt_1""_PASSED"; echo ""; cd /home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/..;  else echo ""; echo "jit_jitt_1""_FAILED"; echo ""; fi; } 2>&1 | tee -a "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/../TKG/test_output_16067716828123/TestTargetResult";
16:28:17  [IncludeExcludeTestAnnotationTransformer] [INFO] exclude file is /home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64_aix_Personal_testList_1/openjdk-tests/TKG/../TestConfig/resources/excludes/latest_exclude_8.txt
16:28:17  ...
16:28:17  ... TestNG 6.14.2 by Cédric Beust (cedric@beust.com)
16:28:17  ...
16:28:17  
Calling Pipeline was cancelled
16:34:50  Sending interrupt signal to process
16:35:04  /bin/sh[4]: 20316372 Terminated
16:35:04  
16:35:04  jit_jitt_1_FAILED
```

It was spinning for 5 mins. then got cancelled. Can we tone this down to something more reasonable, perhaps 64 would be enough?

Reply:
Changed in : cfd7aa6e7164f3ba2a7d30e51a5c78ae657b70a3